Classifier and Summarizer/WebEntityDetection.py

```diff
@@ -0,0 +1,120 @@
+import requests
+import base64
+import json
+
+class ImageDescriptionRetriever:
+  '''
+  A class to retreive the entities present in an image
+  Params:
+    maxEntites : maximum number of entity results to return from the api
+  '''
+  def __init__(self,maxEntities = 3):
+    self.apiUrl = "https://vision.googleapis.com/v1/images:annotate?key=AIzaSyCWFD9gxP4R7J8sWS0ldoVdpbchcB6-hx0" # to do: must hide this securely :/
+    self.maxEntities = maxEntities
+  
+  def findBase64EncodingForImages(self,images:list)->list:
+    '''
+    finds the base-64 encoding and converts it to a utf-8
+    string 
+    Params :
+        images : list of image paths
+    Return type:
+        list : of images encoded in the above format
+    '''
+    base64Encodings = []
+    
+    for imagePath in images:
+      with open(imagePath,'rb') as f:
+        encodedImage = base64.encodebytes(f.read()).decode()
+        base64Encodings.append(encodedImage)
+    
+    return base64Encodings
+
+  def formatSingleRequest(self,data:str) ->dict:
+    '''
+    formats the request as a dict 
+    with the required data and returns it
+    Params :
+      data : a base64-encoded image as a utf-8 string
+    Return type:
+      dict : formatted as shown below
+    '''
+    return {
+      "image":{
+        "content" : data
+      },
+      "features":[
+        {
+          "maxResults": str(self.maxEntities),
+          "type": "WEB_DETECTION"
+        }
+      ],
+      "imageContext": {
+          "webDetectionParams": {
+            "includeGeoResults": "true"
+          }
+      }
+    }
+
+  def prepareRequests(self,encodedImages:list) ->list:
+    '''
+    prepares the requests for making calls
+    to the api. Structures the given images
+    in the format as expected by the API
+    Params :
+      encodedImages : numpy.ndarray of base64 encoded images
+    Return Type:
+    list: of required format
+    '''
+    imageRequests = []
+    
+    for image in encodedImages:
+      imageRequests.append(self.formatSingleRequest(image))
+    
+    return imageRequests
+
+
+  def getBestGuessLabel(self,jsonResponse,imageIndex):
+    ''' return the best guess label '''
```

Comment:
Rather than having the entire "jsonResponse" and "imageIndex" being sent here, should this method be modified to receive only the response specific to this image?

If needed, obtaining the image specific response from the entire response can be separated to another method, but this is optional.

Reply:
Making the image receive only the response specific to the image makes sense. Updated accordingly!